### PR TITLE
feat: Adds dashboard apis for sign in

### PR DESCRIPTION
**Pull request by nkshah2**

### CODE REVIEW

File: api_spec.yaml [hunk 1]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+        '200':
+          description: Indicates success with the status property
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - type: object
+                    properties:
+                      status:
+                        type: string
```
Comment (nkshah2) [inline api_spec.yaml:2971]: Missing status for invalid email

File: api_spec.yaml [hunk 2]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+    post:
+      tags:
+         - Dashboard Recipe
+      operationId: createDashboardUser
+      description: |
+        Delete a mapping between a SuperTokens userId and an external userId.
+      parameters:
+        - $ref: '#/components/parameters/dashboardRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
```
Comment (nkshah2) [inline api_spec.yaml:2948]: Change CDI to say 2.18

File: api_spec.yaml [hunk 3]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+        
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '404':
+          $ref: '#/components/responses/404'
+        
+        '500':
+          $ref: '#/components/responses/500'
+    put:
```
Comment (nkshah2) [inline api_spec.yaml:3018]: We should revoke all sessions for this user when email or password has changed

File: api_spec.yaml [hunk 4]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+                      type: object
+                      properties:
+                        email:
+                          type: string
+                          example: "test@example.com"
+                        userId:
+                          $ref: '#/components/schemas/userId'
+                        timeCreated:
+                          type: number
+                          example: 16382348324
```
Comment (nkshah2) [inline api_spec.yaml:3177]: This should also contain isSuspended

File: api_spec.yaml [hunk 5]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+  
+  
+  /recipe/dashboard/users:
+    get:
+      tags:
+        - Dashboard Recipe
+      operationId: getAllDashboardUsers
+      description: |
+        Get a list of call the dashboard users
+      parameters:
```
Comment (nkshah2) [inline api_spec.yaml:3155]: This should also include rid and cdi version

File: api_spec.yaml [hunk 6]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '404':
+          $ref: '#/components/responses/404'
+        
+        '500':
+          $ref: '#/components/responses/500'
+                        
+  /recipe/dashboard/session/revoke:
```
Comment (nkshah2) [inline api_spec.yaml:3248]: This should be delete

Comment (nkshah2) [reply api_spec.yaml:3248]: Also remove the `/revoke`

File: api_spec.yaml [hunk 7]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+          $ref: '#/components/responses/500'
+    
+                    
+  /recipe/dashboard/signin:
+    post:
+      tags:
+         - Dashboard Recipe
+      operationId: signInDashboardUser
+      description: |
+        Verify a Dashboard user's sessionId
```
Comment (nkshah2) [inline api_spec.yaml:3297]: Wrong description

File: api_spec.yaml [hunk 8]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+        '500':
+          $ref: '#/components/responses/500'
+                        
+  /recipe/dashboard/session/revoke:
+    post:
+      tags:
+         - Dashboard Recipe
+      operationId: RevokeDashboardUsersSession
+      description: |
+        Reokve a Dashboard user's session
```
Comment (nkshah2) [inline api_spec.yaml:3254]:
```suggestion
        Revoke a Dashboard user's session 
```

File: api_spec.yaml [hunk 9]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+                        type: string
+                        enum: ["USER_SUSPENDED_ERROR"]
+                      message:
+                        type: string
+                        example: "You have crossed the free dashboard user limit. Please purchase the dashboard feature or login with a valid user account"
+                  - type: object
+                    properties:
+                      status:
+                        type: string
+                        enum: ["INVAlID_SESSION_ERROR"]
```
Comment (nkshah2) [inline api_spec.yaml:3341]: Should be INVALID_CREDENTIALS_ERROR

File: api_spec.yaml [hunk 10]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+          $ref: '#/components/responses/500'
+  
+  /recipe/dashboard/user/sessions:
+    get:
+      tags:
+        - Dashboard Recipe
+      operationId: getAllSessionsForDashboardUser
+      description: |
+        Get a list of the sessions for the dashboard user 
+      parameters:
```
Comment (nkshah2) [inline api_spec.yaml:3361]: Should also need rid and cdi-version

File: api_spec.yaml [hunk 11]
```diff
@@ -2933,6 +2934,472 @@ paths:
...
+          required: false
+      responses:
+        '200':
+          description: get a list of sessions for dashboard user
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  users:
```
Comment (nkshah2) [inline api_spec.yaml:3377]: Should be called sessions

Comment (nkshah2) [reply api_spec.yaml:3377]: This API should return array in descending order of time created

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**